qiita_db/support_files/patches/python_patches/20.py

```diff
@@ -0,0 +1,204 @@
+                (study_id, column_name, column_type)
+              VALUES (%s, 'physical_specimen_location', 'varchar'),
+                     (%s, 'physical_specimen_remaining', 'bool'),
+                     (%s, 'dna_extracted', 'bool'),
+                     (%s, 'sample_type', 'varchar'),
+                     (%s, 'collection_timestamp', 'timestamp'),
+                     (%s, 'host_subject_id', 'varchar'),
+                     (%s, 'description', 'varchar'),
+                     (%s, 'latitude', 'float8'),
+                     (%s, 'longitude', 'float8')"""
+
+sql_alter = """
+ALTER TABLE qiita.{0} ADD COLUMN physical_specimen_location varchar;
+ALTER TABLE qiita.{0} ADD COLUMN physical_specimen_remaining bool;
+ALTER TABLE qiita.{0} ADD COLUMN dna_extracted bool;
+ALTER TABLE qiita.{0} ADD COLUMN sample_type varchar;
+ALTER TABLE qiita.{0} ADD COLUMN collection_timestamp timestamp;
+ALTER TABLE qiita.{0} ADD COLUMN host_subject_id varchar;
+ALTER TABLE qiita.{0} ADD COLUMN description varchar;
+ALTER TABLE qiita.{0} ADD COLUMN latitude float8;
+ALTER TABLE qiita.{0} ADD COLUMN longitude float8"""
+
+sql_update = """UPDATE qiita.{0}
+                SET physical_specimen_location=%s,
+                    physical_specimen_remaining=%s,
+                    dna_extracted=%s,
+                    sample_type=%s,
+                    collection_timestamp=%s,
+                    host_subject_id=%s,
+                    description=%s,
+                    latitude=%s,
+                    longitude=%s
+                WHERE sample_id=%s"""
+
+for s_id in study_ids:
+    # Get the data from the requried_sample_info
+    data = conn_handler.execute_fetchall(sql_select, (s_id,))
+
+    # Add rows to the study_sample_columns table
+    # 9 -> the number of rows that we are adding, see `sql_cols`
+    conn_handler.add_to_queue(queue_name, sql_cols, [s_id] * 9)
+
+    # Get the dynamic table name
+    table_name = "sample_%d" % s_id
+
+    # Add the columns to the dynamic table
+    conn_handler.add_to_queue(queue_name, sql_alter.format(table_name))
+
+    # Add the values for each sample
+    for sample_data in data:
+        sample_data = dict(sample_data)
+        conn_handler.add_to_queue(
+            queue_name, sql_update.format(table_name),
+            (sample_data['physical_location'],
+             sample_data['has_physical_specimen'],
+             sample_data['has_extracted_data'],
+             sample_data['sample_type'],
+             sample_data['collection_timestamp'],
+             sample_data['host_subject_id'],
+             sample_data['description'],
+             sample_data['latitude'],
+             sample_data['longitude'],
+             sample_data['sample_id']))
+
+# Once all data has been moved, we can drop the columns from the
+# requried_sample_info table; and drop the required_sample_info_status table
+conn_handler.add_to_queue(
+    queue_name,
+    """ALTER TABLE qiita.required_sample_info
```

Comment:
All these drops can be a single SQL statement
```SQL
alter table qiita.required_sample_info drop column collection_timestamp, drop column host_subject_id;
```
just keep adding to the list of drop column statements.